src/Bicep.Core/Semantics/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -504,6 +505,46 @@ private static SyntaxBase StringLiteralFunctionReturnTypeEvaluator(FunctionCallS
             return SyntaxFactory.CreateStringLiteral(stringLiteral.RawStringValue);
         }
 
+        private static TypeSymbol JsonTypeBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        {
+            static TypeSymbol ToBicepType(JToken token)
+                => token switch {
+                    JObject @object => new ObjectType(
+                        "object",
+                        TypeSymbolValidationFlags.Default,
+                        @object.Properties().Select(x => new TypeProperty(x.Name, ToBicepType(x.Value), TypePropertyFlags.ReadOnly | TypePropertyFlags.ReadableAtDeployTime)),
+                        null),
+                    JArray @array => new TypedArrayType(
+                        UnionType.Create(@array.Select(x => ToBicepType(x))),
+                        TypeSymbolValidationFlags.Default),
+                    JValue value => value.Type switch {
+                        JTokenType.String => new StringLiteralType(value.ToString()),
+                        JTokenType.Integer => LanguageConstants.Int,
+                        JTokenType.Float => LanguageConstants.Int,
```

Comment:
I think it would probably make more sense to just return "any" for float.

Reply:
Yeah definitely `any` with a comment why.